File: association-librarian-frontend/src/mappers/userprofile.js
export const mapUserProfileToState = (data) => {
    return {
        fname: data.firstName, mname: data.midName, sname: data.surName, dob: data.dob,
        gzone: data.zone, phone: data.phoneNo, email: data.email, sex: data.sex,
        photo: { url: data.photo, hide: false }, cv: { url: data.cv, hide: false }
    };
}

export const mapStateToUserProfile = (data) => {
    return {
        firstName: data.fname, midName: data.mname, surName: data.sname, dob: data.dob,
        zone: data.gzone, phoneNo: data.phone, email: data.email, sex: data.sex,
        photo: data.photo.url, cv: data.cv.url
    };
}

File: association-librarians-backend/association-librarians/src/main/java/com/librarians/models/UserProfile.java
package com.librarians.models;

import java.util.Date;

import lombok.NonNull;

public record UserProfile (@NonNull String firstName, String midName, @NonNull String surName,
                           @NonNull Date dob, @NonNull GeopoliticalZone zone, @NonNull String email,
                           @NonNull String phoneNo, @NonNull Sex sex, @NonNull String photo, @NonNull String cv) {

    @Override
    public String toString() {
        return email;
    }
}